domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -392,20 +456,58 @@ public Future<Void> closeClient() {
     return clientToClose.close();
   }
 
+  public Future<Void> closeClient(){
+    if (client == null){
+      return Future.succeededFuture();
+    }
+
+    PgPool clientToClose = client;
+    PgPool readClientToClose = client == readClient ? null : readClient;
+    client = null;
+    readClient = null;
+
+    closeClient(clientToClose);
+    return closeClient(readClientToClose);
```

Comment:
```suggestion
    return closeClient(clientToClose).compose(x -> closeClient(readClientToClose));
```

Reply:
Keeping the code as is as using compose fails.. Due to closeAllClients being used without Vert.x context in sync mode - at least in tests.

Reply:
readClientToClose is null if no read client is configured. Therefore it should be swapped:

```
    closeClient(readClientToClose);
    return closeClient(clientToClose);
```